public/app/plugins/datasource/loki/components/LokiExploreQueryEditor.tsx

```diff
@@ -0,0 +1,109 @@
+// Libraries
+import React, { memo, useState, useEffect } from 'react';
+import _ from 'lodash';
+
+// Types
+import { AbsoluteTimeRange, ExploreQueryFieldProps, ExploreMode } from '@grafana/data';
+import { LokiDatasource } from '../datasource';
+import { LokiQuery, LokiOptions } from '../types';
+import { LokiQueryField } from './LokiQueryField';
+import { useLokiSyntax } from './useLokiSyntax';
+import { LokiExploreExtraField } from './LokiExploreExtraField';
+
+type Props = ExploreQueryFieldProps<LokiDatasource, LokiQuery, LokiOptions>;
+
+export const LokiQueryEditor = memo(function LokiQueryEditor(props: Props) {
+  const { query, data, datasource, exploreMode, history, onChange, onRunQuery } = props;
+
+  let absolute: AbsoluteTimeRange;
+  if (data && !_.isEmpty(data.request)) {
+    const { range } = data.request;
+
+    absolute = {
+      from: range.from.valueOf(),
+      to: range.to.valueOf(),
+    };
+  } else {
+    absolute = {
+      from: Date.now() - 10000,
+      to: Date.now(),
+    };
+  }
+
+  const [maxLines, setMaxLines] = useState(query?.maxLines?.toString() || datasource?.maxLines?.toString());
+
+  const { isSyntaxReady, setActiveOption, refreshLabels, ...syntaxProps } = useLokiSyntax(
+    datasource.languageProvider,
+    absolute
+  );
+
+  function onChangeQueryLimit(value: string, override?: boolean) {
+    const { query, onChange, onRunQuery } = props;
+    if (onChange) {
+      const nextQuery = { ...query, maxLines: preprocessMaxLines(value) };
+      onChange(nextQuery);
+
+      if (override && onRunQuery) {
+        onRunQuery();
+      }
+    }
+  }
+
+  function preprocessMaxLines(value: string): number {
```

Comment:
Only issue I have is with the fall-backing here. If I input `0`, `-10`, `aaa` into the input I get `No logs found.Scan for older logs` message. Which technically isn't true, user just set the limit to be nonsensical in essence. Can we instead of fall-back to last sensible (> 0) value in case of those values? Or maybe even not run the query in that case? Also maybe for this simple case (we want just positive integer) a `type=number` input would make sense.

Reply:
User gets exactly the same message when they have at least one valid query with valid limit and they just hide results coming from this query row, so it made sense to do it this way with line limit values lower than or equal to 0. Just to avoid sending unnecessary requests, I changed `onReturnKeyDown` handler to run queries only when `maxLines` is positive, also added a type param to our `LokiExtraField` and set it to be a number.